commands.js

```diff
@@ -701,6 +701,9 @@ var commands = exports.commands = {
 			}
 			return connection.sendTo(target, "|noinit|joinfailed|You do not have permission to join '" + target + "'.");
 		}
+		if (toId(target) in Rooms.aliases) {
+			connection.send(">" + toId(target) + "\n|deinit");
+		}
```

Comment:
I only have one suggestion, which is to move this above `var joinResult = user.joinRoom(targetRoom, connection);`